napari/layers/base/base.py

```diff
@@ -1137,39 +1160,52 @@ def _slice_dims(
             it already has a valid slice ready. False otherwise.
         """
         logger.debug(
-            'Layer._slice_dims: %s, point=%s, ndisplay=%s, order=%s, force=%s',
+            'Layer._slice_dims: %s, point=%s, margin_left=%s, margin_right=%s, ndisplay=%s, order=%s, force=%s',
             self,
             point,
+            margin_left,
+            margin_right,
             ndisplay,
             order,
             force,
         )
-        slice_input = self._make_slice_input(point, ndisplay, order)
+        slice_input = self._make_slice_input(
+            point, margin_left, margin_right, ndisplay, order
+        )
         if force or (self._slice_input != slice_input):
             self._slice_input = slice_input
             self._refresh_sync()
 
     def _make_slice_input(
-        self, point=None, ndisplay=2, order=None
+        self,
+        point=None,
+        margin_left=None,
+        margin_right=None,
```

Comment:
@jni, @andy-sweet, we touched on this before, but maybe not exactly for this context. What do you think? My current thinking is that having a class (or namedtuple) here might look nicer in the signatures and function calls, but it's less intuitive to work with (especially if users end up having to use it).

Reply:
Isn't `Margin` almost `_ThickNDSlice`? Could just create a `_ThickNDSlice` where it's called and pass that through instead of `point`, `margin_left`, and `margin_right`?

Reply:
Both `_make_slice_input` and `_slice_dims` should be executed on the main thread, so if passing through `Dims` is preferred, I think you should go for it.